pkg/ring/lifecycler.go

```diff
@@ -38,6 +38,11 @@ var (
 	}, []string{"op", "status", "name"})
 )
 
+// ReadLifecycler represents the read interface to the lifecycler.
+type ReadLifecycler interface {
```

Comment:
This interface should be defined at the place of consumption (ingestion_rate_strategy.go)